src/table_builder.cc

```diff
@@ -12,7 +12,31 @@ void TitanTableBuilder::Add(const Slice& key, const Slice& value) {
     return;
   }
 
-  if (ikey.type != kTypeValue || value.size() < cf_options_.min_blob_size) {
+  std::shared_ptr<VersionSet> vset;
+  if (ikey.type != kTypeValue &&
+      cf_options_.blob_run_mode == TitanBlobRunMode::kFallback &&
+      (vset = vset_ref_.lock()) != nullptr) {
+    Slice copy = value;
+    uint64_t cf_id = GetTableProperties().column_family_id;
+    BlobIndex index;
+    Status s = index.DecodeFrom(&copy);
+    assert(s.ok());
+
+    BlobRecord record;
+    PinnableSlice buffer;
+
+    auto storage = vset->GetBlobStorage(cf_id).lock();
+
+    ReadOptions options;  // dummy option
+    s = storage->Get(options, index, &record, &buffer);
+    if (s.ok()) {
+      base_builder_->Add(key, record.value);
+      return;
```

Comment:
early return is not very readable. Use if-else instead?
```
if (type == kTypeBlobIndex && blob_run_mode == kFallback) {
   // get blob and convert to normal value
} else if (type == kTypeValue && blob_run_mode == kNormal && value.size() >= min_blob_size) {
   // write blob file and add blob index
} else {
  basd_builder_->Add(key, value)
}
```

Reply:
Here I was actually making sure that index value is not lost if blob storage is corrupted for now. But I guess that's a pointless act too.